desktop/core/src/desktop/js/reactComponents/TaskServer/TaskServer.test.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { render, fireEvent, waitFor } from '@testing-library/react';
+import TaskServer from './TaskServer';
+import { TaskServerResponse, TaskStatus } from './types.ts';
+
+const mockTasks: TaskServerResponse[] = [
+  {
+    taskId: '3cf1c861-faa2-484c-b15a-9a1533b6acf2',
+    dateDone: '2025-03-25T14:30:00Z',
+    status: TaskStatus.Success,
+    result: {
+      username: 'celery_scheduler',
+      taskName: 'tmp_cleanup',
+      parameters: 90,
+      progress: '100%',
+      taskStart: '2025-03-25T04:29:15.674345',
+      taskEnd: '2025-03-25T04:29:15.680075'
+    },
+    children: []
+  },
+  {
+    taskId: 'c098f3d1-c937-4e50-b686-52d677296b95',
+    dateDone: '2025-03-25T15:00:00Z',
+    status: TaskStatus.Running,
+    result: {
+      username: 'celery_scheduler',
+      taskName: 'cleanup_stale_uploads',
+      parameters: 60,
+      progress: '100%',
+      taskStart: '2025-03-24T07:50:10',
+      taskEnd: '2025-03-24T07:50:10.982950'
+    },
+    children: []
+  },
+  {
+    taskId: 'c098f3d1-c937-4e50-b686-52d677296b96',
+    dateDone: '2025-03-25T16:30:00Z',
+    status: TaskStatus.Failure,
+    result: {
+      taskName: 'file_upload',
+      username: 'alex_lee',
+      taskStart: '2025-03-25T16:00:00Z',
+      taskEnd: '',
+      progress: '0%',
+      qqfilename: 'file3.txt',
+      parameters: 0
+    },
+    children: []
+  }
+];
+
+const mockHandleSave = jest.fn();
+jest.mock('../../utils/hooks/useSaveData/useSaveData', () => ({
+  __esModule: true,
+  default: jest.fn(() => ({
+    save: mockHandleSave,
+    loading: false,
+    error: null
+  }))
+}));
+
+jest.mock('../../utils/hooks/useLoadData/useLoadData', () => ({
+  __esModule: true,
+  default: jest.fn(() => ({
+    data: mockTasks,
+    loading: false,
+    error: null
+  }))
+}));
+
+describe('TaskServer Component', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should open schedule task modal on button click', async () => {
+    const { getByRole, getAllByText, getByTestId } = render(<TaskServer />);
+
+    fireEvent.click(getByRole('button', { name: 'Schedule Task' }));
+
+    await waitFor(() => expect(getAllByText('Schedule Task')).toHaveLength(2));
+    await waitFor(() => expect(getByTestId('hue-schedule-task__modal')).toBeInTheDocument());
+  });
+
+  it('should call save API when clicking "Kill Task" button', async () => {
+    const { getAllByRole, getByRole } = render(<TaskServer />);
+
+    fireEvent.click(getAllByRole('checkbox', { name: '' })[0]);
+    fireEvent.click(getByRole('button', { name: 'Kill Task' }));
+
+    await waitFor(() => expect(mockHandleSave).toHaveBeenCalledTimes(1));
```

Comment:
I would expand this a bit and check that the correct API is called with the id of the task